### PR TITLE
Revisit the Taxonomy List widget

### PR DESCRIPTION
**Pull request by benlk**
## Changes

- Remove option to include only chosen terms
- Include ability to exclude chosen terms from returned terms
- Remove ability to output infinite terms, number output must be set and defaults to 5
- Make the taxonomies option a select instead of a text field, preventing empty-taxonomy error
- Creates two sort options:
    - alphabetically by term name
    - newest term first, by descending order of term id (proxy for most-recently-updated)
- Improve docs in the widget's settings
- Add docs in `docs/users/sidebarswidgets.rst`
- Fixes enqueues for the post-type and series taxonomies: https://github.com/INN/Largo/pull/1171#discussion_r57196937 (the previous behavior would register the inactive taxonomies without any labels, leading to spare "tag" labels in taxonomy listings.)

screen shot 2016-03-16 at 11 56 42 pm

## Why

Widget rewrites: to make this widget simpler and more user-friendly. For [HELPDESK-589](http://jira.inn.org/browse/HELPDESK-589).

Taxonomy registration rewrite: The old method of registering the post-type and series was to, if the theme option to register the taxonomy was not active, register them with empty labels and various amounts of 'hiddenness' based on the registration args. This was causing the inactive taxonomies to appear in an earlier version of this pull request with default "Tag" labels: #1173. The fix for this was to instead always register those taxonomies, but if they were not enabled in the theme options, to register them with the arguments `public`, `show_admin_column` and `show_in_nav_menus` set to `false`.

## What else?

This does not include the option to sort by last-updated; WordPress does not provide this option. :disappointed:

### CODE REVIEW

**Comment by benlk** [comment]
Filed as a bug PR because of #1173, improvement because of the helpdesk ticket that started this.

**Comment by aschweigert** [comment]
is that screenshot current? I'd like to remove all the extra inline docs because I think that's overkill if so

**Comment by benlk** [comment]
It's not, my bad. That came from the last PR.

This is the current amount of docs:

screen shot 2016-03-31 at 4 28 50 pm

File: inc/widgets/largo-taxonomy-list.php [hunk 1]
```diff
@@ -53,24 +87,27 @@ function widget( $args, $instance ) {
...
-			$tax_items = get_categories($cat_args);
+			$tax_items = get_categories($term_args);
 
-			switch ($instance['taxonomy']) {
+			switch ( $instance['taxonomy'] ) {
 				case 'series':
-					$this->render_series_list($tax_items, $instance);
+					$this->render_series_list( $tax_items, $instance );
 					break;
 				case 'category':
```
Comment (aschweigert) [inline inc/widgets/largo-taxonomy-list.php]: this seems unnecessary, shouldn't we just use an if/else here to use the one function for series and the other function for everything else?

**Pull request merged**